types/index.d.ts

```diff
@@ -0,0 +1,784 @@
+   * @member {string} [options.delimiter]
+   *   Separator between two Unicode characters in `str` parameter of
+   *   `coffeescript.helpers.replaceUnicodeCodePointEscapes`.
+   * @member {string} [options.error=unicode code point escapes greater than \\u{10ffff} are not allowed]
+   *   Error message if `coffeescript.helpers.replaceUnicodeCodePointEscapes` fails.
+   * @member {string} [options.flags]
+   *   Which flags are present in the regular expression for the replacement operation.
+   *   Must include `u` if provided to support Unicode escapes.
+   * @returns RegExp string with Unicode brace groups in the format `\uxxxx[\uxxxx]`.
+   */
+  replaceUnicodeCodePointEscapes(str: string, options?: ReplaceUnicodeCodePointEscapesOptions): string;
+}
+
+/**
+ * Transpiles CoffeeScript to legacy, high-compatibility ECMAScript versions using Babel.
+ *
+ * @param {string} code CoffeeScript code to be compiled.
+ * @param {coffeescript.Options} [options] CoffeeScript compiler options.
+ * @param {boolean} [options.ast=false] If true, output an abstract syntax tree of the input CoffeeScript source code.
+ * @param {boolean} [options.bare=false] If true, omit a top-level IIFE safety wrapper.
+ * @param {string} [options.filename=index.js] File name to compile.
+ * @param {boolean} [options.header=false] If true, output the `Generated by CoffeeScript` header.
+ * @param {boolean} [options.inlineMap=false] If true, output the source map as a base64-encoded string in a comment at the bottom.
+ * @param {boolean} [options.sourceMap=false] If true, output a source map object with the code.
+ * @param {babel.TransformOptions} [options.transpile={}] Babel transpilation options - see `babel.TransformOptions`.
+ * @returns {babel.BabelFileResult} Babel transpiler result for file.
+ */
+export function transpile(code: string, options?: Options): BabelFileResult;
+
+/**
+ * Compiles CoffeeScript to JavaScript code, then outputs it as a string.
+ *
+ * @param {string} code CoffeeScript code to be compiled.
+ * @param {coffeescript.Options} [options] CoffeeScript compiler options.
+ * @param {boolean} [options.ast=false] If true, output an abstract syntax tree of the input CoffeeScript source code.
+ * @param {boolean} [options.bare=false] If true, omit a top-level IIFE safety wrapper.
+ * @param {string} [options.filename=index.js] File name to compile.
+ * @param {boolean} [options.header=false] If true, output the `Generated by CoffeeScript` header.
+ * @param {boolean} [options.inlineMap=false] If true, output the source map as a base64-encoded string in a comment at the bottom.
+ * @param {boolean} [options.sourceMap=false] If true, output a source map object with the code.
+ * @param {babel.TransformOptions} [options.transpile={}] Babel transpilation options - see `babel.TransformOptions`.
+ * @returns {string} Compiled and unevaluated JavaScript code.
+ */
+export function compile(code: string, options?: Options): string;
```

Comment:
One more issue I found, [according to the docs](https://coffeescript.org/#nodejs-usage), this can also return a `{js, v3SourceMap, sourceMap}` if `options: sourceMap` is `true`

Reply:
Resolved in e55967d @phil294